src/tests/pmfs_demo.c

```diff
@@ -0,0 +1,1528 @@
+			}
+			rc = pmfs_mkdir_start(g_present_pool->p_cmd_pmfs,
+					      g_present_pool->p_cmd_obj, argv2, O_RDWR);
+			if (rc < 0) {
+				D_PRINT("mkdir failed\r\n");
+				continue;
+			}
+			if (demo) {
+				folders++;
+				if (folders > 20) {
+					folders = 0;
+					D_PRINT("demo app cmd last = %-10.6f sec \r\n",
+							*(g_vfca->duration));
+					D_PRINT("demo app cmd update = %-10.3f IO/sec \r\n",
+							2/(*(g_vfca->duration)));
+					demo = false;
+					continue;
+				}
+				sprintf(argv1, "%s", "demo");
+				goto demo;
+			}
+
+			continue;
+		} else if (check_cmd(argv1, "pwd")) {
+			if (g_present_pool->p_parent_cmd_obj == NULL) {
+				D_PRINT("\r/\n");
+			} else {
+				D_PRINT("\rneed further support\n");
+			}
+			continue;
+		} else if (check_cmd(argv1, "ls")) {
+			if (g_present_pool->p_cmd_pmfs == NULL) {
+				D_PRINT("no mount point\r\n");
+				continue;
+			}
+
+			uint32_t nr = 0;
+
+			rc = pmfs_listdir_start(g_present_pool->p_cmd_pmfs,
+						g_present_pool->p_cmd_obj, &nr);
+			if (rc < 0) {
+				D_PRINT("list dir failed\r\n");
+				continue;
+			}
+			D_PRINT("----total %d files---\r\n", nr);
+			continue;
+		} else if (check_cmd(argv1, "lspmem")) {
+			check_filesystem();
+			continue;
+		} else if (check_cmd(argv1, "rm ")) {
+			pmfs_path_root_reactor(argv2, 1);
+			daos_obj_id_t oid;
+
+			rc = pmfs_remove_start(g_present_pool->p_cmd_pmfs,
+					       g_present_pool->p_cmd_obj, argv2,
+						true, &oid);
+			if (rc != 0)
+				D_PRINT("pmfs remove start failed\r\n");
+			D_PRINT("pmfs remove done \r\n");
+			continue;
+		} else if (check_cmd(argv1, "cd ") && check_cmd(argv2, "../")) {
+			if (!g_present_pool->p_cmd_obj || !g_present_pool->p_cmd_pmfs) {
+				continue;
+			}
+			mode_t mode;
+			struct pmfs_obj *tmp_obj = g_present_pool->p_cmd_obj;
+
+			rc = pmfs_lookup_start(g_present_pool->p_cmd_pmfs, g_name, O_RDONLY,
+					       &tmp_obj, &mode, NULL);
+			if (rc != 0  || !tmp_obj) {
+				D_PRINT("no such file or dir %s\r\n", argv2);
+				continue;
+			}
+
+			rc = pmfs_release_start(tmp_obj);
+			if (rc != 0) {
+				D_PRINT("pmfs release tmp_obj start failed\r\n");
+			continue;
+			}
+			D_PRINT("pmfs release obj %p\r\n", tmp_obj);
+			g_present_pool->p_cmd_obj = g_present_pool->p_parent_cmd_obj;
+			g_name = NULL;
+			continue;
+		} else if (check_cmd(argv1, "stat ") || check_cmd(argv1, "s ") ||
+				check_cmd(argv1, "stat")) {
+			rc = pmfs_stat_start(g_present_pool->p_cmd_pmfs, g_present_pool->p_cmd_obj,
+					     argv2, &stbuf);
+			if (rc != 0) {
+				D_PRINT("pmfs stat start failed\r\n");
+				continue;
+			}
+			D_PRINT("total size =%ld", stbuf.st_size);
+			D_PRINT("\t  File type and mode  =%x \r\n", stbuf.st_mode);
+			continue;
+		} else if (check_cmd(argv1, "cd ")) {
+			g_present_pool->p_parent_cmd_obj = g_present_pool->p_cmd_obj;
+			rc = pmfs_open_start(g_present_pool->p_cmd_pmfs, g_present_pool->p_cmd_obj,
+					     argv2, S_IFDIR, O_RDWR,
+			     		     1024, NULL, &g_present_pool->p_cmd_obj);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs